streams/src/main/java/org/apache/kafka/streams/kstream/internals/InternalStreamsBuilder.java

```diff
@@ -252,6 +288,111 @@ public void buildAndOptimizeTopology() {
         }
     }
 
+    private void maybePerformOptimizations(final Properties props) {
+
+        if (props != null && props.getProperty(StreamsConfig.TOPOLOGY_OPTIMIZATION).equals(StreamsConfig.OPTIMIZE)) {
+            LOG.debug("Optimizing the Kafka Streams graph for repartition nodes");
+            maybeOptimizeRepartitionOperations();
+        }
+    }
+
+    private void maybeOptimizeRepartitionOperations() {
+        StreamsGraphNode optimizedSingleRepartition = null;
```

Comment:
This doesn't need to be declared outside the loop (it can be final at the assignment).

Reply:
ack